### ISSUE TITLE
dotnet45 registry error

### ISSUE THREAD

**Issue by Zemogiter** [PhoenicisOrg/scripts#720]
When put inside any script, dotnet45 errors with following message:
```
java.lang.IllegalArgumentException: Error while parsing the registry
	at org.phoenicis.win32/org.phoenicis.win32.registry.RegistryParser.parseFile(RegistryParser.java:77)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1584$1109A$\^eval\_$cu1$restOf.regedit#fetchValue([engines, wine, plugins, regedit]:52)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1582$432AA$\^eval\_.windowsVersion([engines, wine, plugins, windows_version]:15)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1581$511$\^eval\_$cu1$restOf.dotnet45([engines, wine, verbs, dotnet45]:20)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1574$812AA$\^eval\_.run#L:22(<eval>:23)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1494$553$\^eval\_$cu1$restOf.go([engines, wine, quick_script, installer_script]:75)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1463$402$\^eval\_.run(<eval>:13)
	at jdk.nashorn.javaadapters.org_phoenicis_scripts_Installer.run(Unknown Source)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1462$\^eval\_$cu1$restOf.:program(<eval>:35)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:655)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:513)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:527)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:456)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:413)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts/org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.FileNotFoundException: /home/jonasz/.Phoenicis/containers/wineprefix/The Sims 3/user.reg (Nie ma takiego pliku ani katalogu)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at java.base/java.io.FileReader.<init>(FileReader.java:75)
	at org.phoenicis.win32/org.phoenicis.win32.registry.RegistryParser.parseFile(RegistryParser.java:44)
	... 22 more
```
Installation via winetricks works fine (tested on Anno 2070 and Sims 3 scrips). The error states that file user.reg doesn't exist but it's there when I opened the wineprefix.

**Comment by ImperatorS79** [comment]
All the dotnet scripts have to be updated to match winetricks. I do not have time to check it currently unfortunately.

**Comment by plata** [comment]
What exactly means "match winetricks"?

**Comment by plata** [comment]
Well, if really deleting the registry changes is the solution, it's an easy fix.

**Comment by ImperatorS79** [comment]
I mean those verbs are based on the winetricks ones, and there are differences (I think) between the dotnet verbs here and in winetricks.

**Comment by Zemogiter** [comment]
For some weird reason dotnet461 works in Space Engineers script but when I tried in in Sims 3 it gave me the exact same error.

**Comment by plata** [comment]
That's strange. Can you investigate why this happens?

**Comment by Zemogiter** [comment]
@plata ok so here is my investigation:
I've tried to make both scripts as alike as possible in order to see what might cause this error. First I changed wine version and distribution in Sims 3 script - it still generates this error. Then I've switched form `dotnet45` to `dotnet461` and the outcome is the same. But when I've changed the architecture on Sims 3 script form x86 to 64bit the error was gone. Did MS dropped support for x32 systems in their .net framework installers without telling anyone? Because it looks like they did.
Now I'll do the same but with all changes to Sims 3 script reverted expect for the architecture.
